index.js

```diff
@@ -36,12 +36,29 @@ function parse(topic) {
 	var result = {
 		regex: make_regex(tokens),
 		getParams: make_pram_getter(tokens),
-		topic: make_clean_topic(tokens)
+		topic: make_clean_topic(tokens),
+		format: make_formatter(topic)
 	};
 	result.exec = exec.bind(result);
 	return result;
 };
 
+function make_formatter(topic) {
+	function formatter(params) {
+		var tokens = tokenize(topic)
+		for (var i = 0; i < tokens.length; i++) {
+			if (tokens[i][0] == '+' || tokens[i][0] == '#') {
+				let paramName = tokens[i].slice(1)
+				tokens[i] = params[paramName]
+			}
+		}
+		return tokens.join('/')
```

Comment:
Should there maybe be a `tokens.flat()` call so that `#` parameters can be arrays?

Reply:
Good point, I wasn't considering the `#` wildcard. Done.